### DISCUSSION TITLE
Is there any way to identifiy whether a caught error is in a scope outside the catch block?

### DISCUSSION THREAD

**Discussion by mjr4077au** [PowerShell/PowerShell#21020] [Q&A]
Possibly not worded well, but I'll try to explain.

I'm currently using PowerShell transcriptions for logging purposes. When a function throws a terminating error, the transcription will show the following for every function in the call stack until it hits the catch block: `PS>TerminatingError(): "test throw"`.

If I'm catching a throw within the same scope (`try {throw 'test throw'} catch {...}`), there's no `PS>TerminatingError(): "test throw"` line in the transcription, which means I need to write one out myself.

The problem here is for situations where if I just blindly write out the message myself, I'll get a look that's like:
```
PS>TerminatingError(): "test throw"
test throw
```
Ideally I'd like to not write my own line if I know the transcriber has caught a terminating error and logged it as `PS>TerminatingError(): "test throw"`.

Is there any robust way to check for this? This is mainly for my lynch-pin main try/catch block. Moving the catches closer isn't necessarily an option.

**Comment by fMichaleczek** [comment]
@mjr4077au stackoverflow.com is in my opinion a better place to ask your questions.

I'm not sure I understand your need but it appears that you don't know the $PSCmdlet variable.

Hope this link can help you.
https://stackoverflow.com/questions/49204918/difference-between-throw-and-pscmdlet-throwterminatingerror

**Reply by mjr4077au** [reply]
Thanks mate, I appreciate your time, and yeah you're probably right about Stack Overflow as well.

I do know about the `$PSCmdlet` automatic variable, but I don't always have CmdletBinding on functions and calling `ThrowTerminatingError()` means I have my own `ErrorRecord` objects vs. just throwing a string. I'll test whether a throw via `ThrowTerminatingError()` changes the printout in the transcription though.

**Reply by fMichaleczek** [reply]
Transcript was made mostly for JEA, because it's a mandatory feature to ensure security audit. 

So don't be surprised if the features aren't complete as a Logger. 

Here is chatgpt's answer on the difference. And I think your usage is not the right one. (throw is not a command) 

![image](https://github.com/PowerShell/PowerShell/assets/6187529/0fdd082b-8625-4e40-8f5d-39e198f6ae2c)

If you want a logger, look on the Gallery, personally I really like PoShLog which is a wrapper of the Serilog library. It's very powerful and configurable. You can even enrich your logs with Error Records.

https://github.com/PoShLog/PoShLog/blob/master/src/functions/enrichers/Add-EnrichWithErrorRecord.ps1

**Reply by mjr4077au** [reply]
I like the look of PoShLog a lot, however it's not been in active development for two years now and I fear getting vested in it, only to encounter an issue that necessitates me forking it and maintaining it.

What I've got now is relatively light and sufficient for my needs, I think if those messages in the transcript bother me enough I can just make my `Write-LogMessage` function output text using `Out-File -Append` or something similar.
